health_check/contrib/mail/backends.py

```diff
@@ -0,0 +1,28 @@
+import logging
+
+from django.core.mail import get_connection
+
+from health_check.backends import BaseHealthCheckBackend
+from health_check.conf import HEALTH_CHECK
+from health_check.exceptions import ServiceUnavailable
+
+logger = logging.getLogger(__name__)
+
+
+class MailHealthCheck(BaseHealthCheckBackend):
+    """Check that mail backend is working."""
+
+    def check_status(self) -> None:
+        """Open and close connection email server."""
+        try:
+            connection = get_connection(fail_silently=False)
+            connection.timeout = HEALTH_CHECK.get("MAIL_TIMEOUT", 15)
+            logger.debug("Trying to open connection to mail backend.")
+            connection.open()
+            connection.close()
+            logger.debug("Connection established. Mail backend is healthy.")
+        except Exception as error:
```

Comment:
This is an overly long try block. What's your point of failure that indicates a service being down? I presume the connect statement?

You might also want to consider sending an `HELO` command to the SMPT server.

Lastly, please consider moving the close statement into the `finally` block. You don't want any unclosed connections.